src/components/tl/ucp/allreduce/allreduce.c

```diff
@@ -46,3 +50,65 @@ ucc_status_t ucc_tl_ucp_allreduce_knomial_init(ucc_base_coll_args_t *coll_args,
 out:
     return status;
 }
+
+ucc_status_t
+ucc_tl_ucp_allreduce_sliding_window_init(ucc_base_coll_args_t *coll_args,
+                                         ucc_base_team_t      *team,
+                                         ucc_coll_task_t     **task_h)
+{
+    ucc_tl_ucp_team_t       *tl_team = ucc_derived_of(team, ucc_tl_ucp_team_t);
+    ucc_rank_t               size    = (ucc_rank_t)team->params.size;
+    ucc_status_t             status  = UCC_OK;
+    ucc_tl_ucp_task_t       *task;
+    ucc_ee_executor_params_t params;
+
+    ALLREDUCE_TASK_CHECK(coll_args->args, tl_team);
+
+    if (!(coll_args->args.mask & UCC_COLL_ARGS_FIELD_GLOBAL_WORK_BUFFER)) {
+        tl_error(UCC_TL_TEAM_LIB(tl_team),
+                 "global work buffer not provided nor associated with team");
+        status = UCC_ERR_NOT_SUPPORTED;
+        goto out;
+    }
+    if (coll_args->args.mask & UCC_COLL_ARGS_FIELD_FLAGS) {
+        if (!(coll_args->args.flags & UCC_COLL_ARGS_FLAG_MEM_MAPPED_BUFFERS)) {
+            tl_error(UCC_TL_TEAM_LIB(tl_team),
+                     "non memory mapped buffers are not supported");
+            status = UCC_ERR_NOT_SUPPORTED;
+            goto out;
+        }
+    }
+    task                 = ucc_tl_ucp_init_task(coll_args, team);
+    *task_h              = &task->super;
+    task->super.post     = ucc_tl_ucp_allreduce_sliding_window_start;
+    task->super.progress = ucc_tl_ucp_allreduce_sliding_window_progress;
+    task->super.finalize = ucc_tl_ucp_allreduce_sliding_window_finalize;
+
+    ucc_tl_ucp_allreduce_sw_global_work_buf_info *gwbi_p =
+        (ucc_tl_ucp_allreduce_sw_global_work_buf_info *)TASK_ARGS(task).global_work_buffer;
+
+    task->allreduce_sliding_window.src_rkeys    = &gwbi_p->src_rkeys[size * gwbi_p->tid];
+    task->allreduce_sliding_window.dst_rkeys    = &gwbi_p->dst_rkeys[size * gwbi_p->tid];
+    task->allreduce_sliding_window.host_eps     = &gwbi_p->host_eps[size * gwbi_p->tid];
+    task->allreduce_sliding_window.worker       = gwbi_p->ucp_thread_workers[gwbi_p->tid];
+    task->allreduce_sliding_window.sbufs        = gwbi_p->sbufs;
+    task->allreduce_sliding_window.rbufs        = gwbi_p->rbufs;
+    task->allreduce_sliding_window.pipe         = &gwbi_p->pipes[gwbi_p->tid];
+    task->allreduce_sliding_window.num_get_bufs = gwbi_p->num_bufs - 1;
+    task->allreduce_sliding_window.window_size  = gwbi_p->window_size;
+    task->allreduce_sliding_window.tid          = gwbi_p->tid;
+    task->allreduce_sliding_window.nthreads     = gwbi_p->nthreads;
+    task->allreduce_sliding_window.put_requests = gwbi_p->pipes[gwbi_p->tid].put_requests;
+
+    params.mask    = UCC_EE_EXECUTOR_PARAM_FIELD_TYPE;
+    params.ee_type = UCC_EE_CPU_THREAD;
+    status         = ucc_ee_executor_init(&params,
```

Comment:
If I don't initialize an executor here, then there is a segfault in the reduction that indicates it's trying to use an executor that doesn't exist. I have not debugged the issue further since initializing one here seemed to fix the problem. If I remember correctly, the reduction would call a function that looks for an executor on the task.super object, going upwards until it finds one. In this case, it never found one.